admin/includes/functions/general.php

```diff
@@ -246,8 +246,10 @@ function zen_get_category_tree($parent_id = '0', $spacing = '', $exclude = '', $
 
 ////
 // products with name, model and price pulldown
-  function zen_draw_products_pull_down($name, $parameters = '', $exclude = '', $show_id = false, $set_selected = false, $show_model = false, $show_current_category = false) {
-    global $currencies, $db, $current_category_id;
+function zen_draw_products_pull_down($name, $parameters = '', $exclude = '', $show_id = false, $set_selected = false, $show_model = false, $show_current_category = false) {
+    global $currencies, $db, $current_category_id, $prev_next_order;
+
+    $order_by = $prev_next_order ?? ' ORDER BY products_name';
```

Comment:
1. While I'd love to use `??` all over the place, at this point we're still trying to keep the code compatible with PHP 5.5/5.6. So, you could rework this to use `?:` or a full on `isset/isnull ? use : else` ternary.

2. It's not save to trust `$prev_next_order` here, without sanitizing it first. Else you risk introducing SQL Injections. You could wrap the whole assignment with `$db->prepare_input()`.

Reply:
ok.
regarding the sanitation by $db->prepare_input.

For this clause
`$prev_next_order = ' ORDER BY LPAD(p.products_id,11,"0")';`
it produces
`
ORDER BY LPAD(p.products_id,11,\"0\")`

which chokes the query.

Is that LPAD necessary? It can't be just p.products_id?

Reply:
I believe the padding is there to force product ID sorting to be truly numeric, like 1,2,3,4,10,11,100,1000 instead of 1,10,11,100,1000,2,3,4

Reply:
Either way seems to work as a natural sort
![orderby1](https://user-images.githubusercontent.com/4391026/65723993-caedee00-e0af-11e9-8ea6-82d5d04e0786.gif)

![orderby2](https://user-images.githubusercontent.com/4391026/65724063-e8bb5300-e0af-11e9-9d74-b1964d7ae27c.gif)

and in my shop, for the admin category-product listing I have long had a sort by product_id and that also gives a natural sort.

Reply:
So the LPAD can be removed...?

Reply:
I've removed the LPAD in `/admin/includes/modules/products_prev_next.php`.

You still need to sanitize the variable, since you're accepting it in from global scope without any validation.

Reply:
ok